### ISSUE TITLE
Mocking a constructor function with no prototypes does not correctly mock static methods

### ISSUE THREAD

**Issue by NervosaX** [testdouble/testdouble.js#208]
I have the following function:

```javascript
function output() {}
output.success = function() {};
output.error = function() {};
```

I pass this to function as an argument, and want to test that `output.success` or error was called.

However when I do the following:
```javascript
td.constructor(output)
```

I receive the following error:

```console
TypeError: output.error is not a function
```

If instead I add a random (unused) prototype to the function:

```
function output() {}
output.prototype.none = function() {};
output.success = function() {};
output.error = function() {};
```

success and error are successfully mocked and my tests work.

**Comment by searls** [comment]
Good catch. In this case I can't imagine why you wouldn't _want_ to be using `td.function(output)`, since that'll also create fakes for `success` & `error`, but I'll die to defend your right to use `td.constructor()` anyway.

Fixed in 09c08a774d27a1f9dd7777025e2f366a0529361f

**Comment by NervosaX** [comment]
That's... also a good call. Thankyou for fixing it anyway. :)
